dask_sql/mappings.py

```diff
@@ -11,6 +11,20 @@
 
 logger = logging.getLogger(__name__)
 
+
+def decimal_or_float():
+    """
+    Return a decimal type if available otherwise fallback to float
+    """
+    try:
+        from cudf import Decimal64Dtype
+
+        return Decimal64Dtype(1)
```

Comment:
cc @ayushdg @jdye64 do we have any preference for what the default decimal type should be here? This is essentially what would get used if we were to do something like

```python
from dask_sql import Context
from dask.datasets import timeseries

c = Context()
c.create_table("df", timeseries(), gpu=True)

c.sql("select cast(x as decimal) from df")
```

My first thought is that we should use `Decimal128Dtype` so that we can have the same maximum precision as the corresponding datafusion type; from there I'm not really sure what the precision/scale should be set to, though I think it should be changed from what it currently is since this results in floats getting rounded to 0.

Also interested in what our expectation is for casting here? Like if we attempt to cast a CPU column to decimal and have cuDF installed, should we create a cuDF dataframe to do this casting or just fail immediately? Currently we get an error deep in the pandas stack that isn't immediately clear